src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalysisState.AnalyzerStateData.cs

```diff
@@ -56,7 +57,7 @@ public void ResetToReadyState()
 
             public virtual void Free()
             {
-                this.StateKind = StateKind.ReadyToProcess;
+                Debug.Assert(StateKind == StateKind.ReadyToProcess);
```

Comment:
Note that this will change to `Debug.Assert(StateKind == StateKind.AttemptingToComplete);` after we implement https://github.com/dotnet/roslyn/issues/65330